File: lc/c/016.c
bool canJump(int* nums, int numsSize){
    int farthest = 0, i = 0, j = numsSize-1;
    while (i <= farthest && farthest < j) {
        farthest = i+nums[i] > farthest ? i+nums[i] : farthest;
        i++;
    }
    return farthest >= j;
}

File: lc/py/016.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        farthest, i, j = 0, 0, len(nums)-1
        while i <= farthest and farthest < j:
            farthest = max(farthest, i+nums[i])
            i += 1
        return farthest >= j
